compiler/rustc_typeck/src/check/check.rs

```diff
@@ -1380,18 +1380,24 @@ fn check_enum<'tcx>(
                 Some(ref expr) => tcx.hir().span(expr.hir_id),
                 None => v.span,
             };
-            let display_discr = display_discriminant_value(tcx, v, discr.val);
-            let display_discr_i = display_discriminant_value(tcx, variant_i, disr_vals[i].val);
-            struct_span_err!(
+            let display_discr = format_discriminant_overflow(discr, tcx, v);
+            let display_discr_i = format_discriminant_overflow(disr_vals[i], tcx, variant_i);
+            let no_disr = v.disr_expr.is_none();
+            let mut err = struct_span_err!(
                 tcx.sess,
                 span,
                 E0081,
-                "discriminant value `{}` already exists",
-                discr.val,
-            )
-            .span_label(i_span, format!("first use of {display_discr_i}"))
-            .span_label(span, format!("enum already has {display_discr}"))
-            .emit();
+                "discriminant value `{}` assigned more than once",
+                discr,
+            );
+
+            err.span_label(i_span, format!("first assignment of {display_discr_i}"));
+            err.span_label(span, format!("second assignment of {display_discr}"));
```

Comment:
This is kinda redundant, since we're already pointing out the span in the error on line 1388. Can you remove this?

Reply:
Not sure if i'm following, you mean remove these two lines?
```
            err.span_label(i_span, format!("first assignment of {display_discr_i}"));
            err.span_label(span, format!("second assignment of {display_discr}"));
```

Reply:
Sorry, just the second one. Github comments didn't just highlight one line.